ios/brave-ios/Sources/Brave/Frontend/UserContent/UserScripts/Scripts_Dynamic/ScriptHandlers/Paged/BraveSearchResultAdScriptHandler.swift

```diff
@@ -0,0 +1,124 @@
+// Copyright 2024 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+import WebKit
+import BraveCore
+import os.log
+
+class BraveSearchResultAdScriptHandler: TabContentScript {
+  private struct SearchResultAdResponse: Decodable {
+    struct SearchResultAd: Decodable {
+      let creativeInstanceId: String
+      let placementId: String
+      let creativeSetId: String
+      let campaignId: String
+      let advertiserId: String
+      let landingPage: URL
+      let headlineText: String
+      let description: String
+      let rewardsValue: String
+      let conversionUrlPatternValue: String?
+      let conversionAdvertiserPublicKeyValue: String?
+      let conversionObservationWindowValue: Int?
+    }
+
+    let creatives: [SearchResultAd]
+  }
+
+  fileprivate weak var tab: Tab?
+
+  private static let secondsInDay = 24 * 60 * 60
```

Comment:
We have some shorthands if you want (`secondsInDay  = 1.day`)

Reply:
changed time interval calculation to:
`let timeInterval = TimeInterval(conversionObservationWindowValue) * 1.days`